lib/tenstorrent/bh_chip/bh_chip.c

```diff
@@ -126,13 +127,47 @@ int therm_trip_gpio_setup(struct bh_chip *chip)
 	if (ret != 0) {
 		return ret;
 	}
+	gpio_init_callback(&chip->therm_trip_cb, therm_trip_detected,
+			   BIT(chip->config.therm_trip.pin));
+	ret = gpio_add_callback_dt(&chip->config.therm_trip, &chip->therm_trip_cb);
+	if (ret != 0) {
+		return ret;
+	}
 	ret = gpio_pin_interrupt_configure_dt(&chip->config.therm_trip, GPIO_INT_EDGE_TO_ACTIVE);
+
+	return ret;
+}
+
+void pgood_change_detected(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	struct bh_chip *chip = CONTAINER_OF(cb, struct bh_chip, pgood_cb);
+
+	/* Sample PGOOD to see if it rose or fell */
+	/* TODO: is this reliable enough? */
```

Comment:
I think this ok- we could also set the interrupt only for PGOOD falling, then set an interrupt for PGOOD rising only once PGOOD had actually fallen